lib/cli/diskpart.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const os = require('os')
+const fs = require('fs')
+const path = require('path')
+const crypto = require('crypto')
+const childProcess = require('child_process')
+const debug = require('debug')('etcher:cli:diskpart')
+const Promise = require('bluebird')
+const retry = require('bluebird-retry')
+
+const TMP_RANDOM_BYTES = 6
+const DISKPART_DELAY = 2000
+const DISKPART_RETRIES = 5
+
+/**
+ * @summary Generate a tmp filename with full path of OS' tmp dir
+ * @function
+ * @private
+ *
+ * @param {String} extension - temporary file extension
+ * @returns {String} filename
+ *
+ * @example
+ * const filename = tmpFilename('.sh');
+ */
+const tmpFilename = (extension) => {
+  const random = crypto.randomBytes(TMP_RANDOM_BYTES).toString('hex')
+  const filename = `etcher-diskpart-${random}${extension}`
+  return path.join(os.tmpdir(), filename)
```

Comment:
Should this be using the new `CONSTANTS.TMP_DIRECTORY` ?

Reply:
There is no `CONSTANTS.TMP_DIRECTORY` in the cli or anywhere else yet; that might be something to do & consolidate later.

Reply:
Ah, I was getting confused with the other PR that I was reviewing at the same time https://github.com/resin-io/etcher/pull/1843/files#diff-0472e5ca31ec513d2fe254fad48d1965R34